### DISCUSSION TITLE
Dynamically generating the access token

### DISCUSSION THREAD

**Discussion by anmoluppal** [tweepy/tweepy#1789] [Questions] [answered]
I am trying to login multiple users through a single app, each user has his own consumer_key and consumer_secret, and i need the access token to login the user through my twitter app, I searched documentation and found the following steps:

-Get a request token from twitter.

-Redirect user to twitter.com to authorize our application

-If using a callback, twitter will redirect the user to us. Otherwise the user must manually supply us with the verifier code.

-Exchange the authorized request token for an access token.

I am stuck at the second and third step. The documentation says to do : 
`# Example using callback (web app)`
`verifier = request.GET.get('oauth_verifier')`

Can someone paste Python code snippet to do the same ? I am getting much confused in these steps.

**Comment by ydaniels** [comment]
Yea you have to get url from twitter through tweepy , then the user would open the url and authorize your client. twitter would give the user a code which you would pass in your app to get the token

**Comment by Harmon758** [comment, marked as answer]
I've rewritten the [authentication documentation about this](https://tweepy.readthedocs.io/en/v4.5.0/authentication.html#legged-oauth) to be clearer now.
